javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -97,17 +98,11 @@ abstract class AbstractMap[K, V] protected () extends java.util.Map[K, V] { self
     throw new UnsupportedOperationException()
 
   def remove(key: Any): V = {
-    // scalastyle:off return
     val iter = entrySet.iterator
-    while (iter.hasNext) {
-      val item = iter.next()
-      if (key === item.getKey) {
-        iter.remove()
-        return item.getValue
-      }
+    iter.find(_.getKey === key).fold(null.asInstanceOf[V]) {
+      iter.remove()
```

Comment:
Er ... I didn't notice before, but this is really bad! The double usage of `iter`, once to convert scala Iterator and `find`, then to call `remove()`, and relying on the fact that the scala Iterator view will leave `iter` in a state where `remove()` will remove the good element, er ... that's really dangerous.

The previous implementation was better, I think.